addons/account_sequence_fix/models/account_move.py

```diff
@@ -0,0 +1,54 @@
+import re
+from itertools import groupby
+
+from odoo import models, fields, api
+
+
+class SequenceMixin(models.AbstractModel):
+    _inherit = 'sequence.mixin'
+
+    sequence_prefix = fields.Char(compute='_compute_split_sequence', store=True)
+    sequence_number = fields.Integer(compute='_compute_split_sequence', store=True)
+
+    def __init__(self, pool, cr):
+        api.depends(self._sequence_field)(type(self)._compute_split_sequence)
+        return super().__init__(pool, cr)
+
+    def _compute_split_sequence(self):
+        regex = self._sequence_fixed_regex
+        regex_seq = re.sub(r"\?P<\w+>", "?:", regex.replace(r"?P<seq>", ""))  # make the seq the only matching group
+        regex_prefix = re.sub(r"\?P<\w+>", "?:", regex.replace(r"?P<prefix1>", ""))  # make prefix1 the only matching group
+
+        self.flush(['name'])
+        self.env.cr.execute("""
+               UPDATE {table}
+                  SET sequence_number = ('0' || (regexp_match({field}, %s))[1])::integer,
+                      sequence_prefix = (regexp_match({field}, %s))[1]
+                WHERE id IN %s
+            RETURNING id, sequence_number, sequence_prefix
+        """.format(
+            field=self._sequence_field,
+            table=self._table
+        ), (regex_seq, regex_prefix, tuple(self.ids)))
+        result = {r[0]: r for r in self.env.cr.fetchall()}
+        for record in self:
+            record.sequence_number = result[record.id][1]
+            record.sequence_prefix = result[record.id][2]
```

Comment:
We need to mark somehow that we don't need to update the database here. I don't know if its feasible?
Or is the whole idea of updating through SQL (for perfs on install of module in existing db) instead of through the ORM a very bad idea?

Reply:
I did the test in master on my `populate` branch with a medium size (~34000 `account.move`)
It takes ~3s to update compute the values at install time when using SQL and 
```python3
del self.env.all.towrite[self._name][record.id]['sequence_number']
del self.env.all.towrite[self._name][record.id]['sequence_prefix']
```
and it takes ~13s with the ORM. I guess it is indeed acceptable for a typical business flow, but it wouldn't hurt to have the time reduced in migrations.